Common/Orders/Fees/InteractiveBrokersTieredFeeModel.cs

```diff
@@ -0,0 +1,287 @@
+            if (_lastOrderTime.Month != order.Time.Month && _lastOrderTime != DateTime.MinValue)
+            {
+                _monthlyTradeVolume = _monthlyTradeVolume.ToDictionary(kvp => kvp.Key, _ => 0m);
+            }
+            // Reprocess the rate schedule based on the current traded volume in various assets.
+            ReprocessRateSchedule(_monthlyTradeVolume[SecurityType.Equity], _monthlyTradeVolume[SecurityType.Future], _monthlyTradeVolume[SecurityType.Forex],
+                _monthlyTradeVolume[SecurityType.Option], _monthlyTradeVolume[SecurityType.Crypto]);
+
+            // Option exercise for equity options is free of charge
+            if (order.Type == OrderType.OptionExercise)
+            {
+                var optionOrder = (OptionExerciseOrder)order;
+
+                // For Futures Options, contracts are charged the standard commission at expiration of the contract.
+                // Read more here: https://www1.interactivebrokers.com/en/index.php?f=14718#trading-related-fees
+                if (optionOrder.Symbol.ID.SecurityType == SecurityType.Option)
+                {
+                    return OrderFee.Zero;
+                }
+            }
+
+            var quantity = order.AbsoluteQuantity;
+            decimal feeResult;
+            string feeCurrency;
+            var market = security.Symbol.ID.Market;
+            switch (security.Type)
+            {
+                case SecurityType.Forex:
+                    // Update the monthly value traded
+                    _monthlyTradeVolume[SecurityType.Forex] += InteractiveBrokersFeeHelper.CalculateForexFee(security, order, _forexCommissionRate, _forexMinimumOrderFee, out feeResult, out feeCurrency);
+                    break;
+
+                case SecurityType.Option:
+                case SecurityType.IndexOption:
+                    var orderPrice = InteractiveBrokersFeeHelper.CalculateOptionFee(security, order, quantity, market, _optionFee, out feeResult, out feeCurrency);
+                    // Regulatory Fee: Options Regulatory Fee (ORF) + FINRA Consolidated Audit Trail Fees
+                    var regulatory = _optionExchangesOrfFee.Contains(market) ?
+                        (0.01915m + 0.0048m) * quantity :
+                        0.0048m * quantity;
+                    // Transaction Fees: SEC Transaction Fee + FINRA Trading Activity Fee (only charge on sell)
+                    var transaction = order.Direction == OrderDirection.Sell ?
+                        0.0000278m * Math.Abs(order.GetValue(security)) + 0.00279m * quantity :
+                        0m;
+                    // Clearing Fee
+                    var clearing = Math.Min(0.02m * quantity, 55m);
+
+                    feeResult += regulatory + transaction + clearing;
+
+                    // Update the monthly value traded
+                    _monthlyTradeVolume[SecurityType.Option] += quantity * orderPrice;
```

Comment:
Whops don't think this state is correct, the lean engine can and will call `GetOrderFee` multiple times for the same order, when submitting later when processing in the backtesting brokerage, can even be more times in cases of setHoldings where it can call it N times to get the best position size. Maybe something that could be done is look at the order status, if submitted use it to account for the monthly volume, should also keep a guard not to process the same order id multiple times I think, but needs more testing and asserting to be sure it's okay
Seems should make the monthly traded volume public so it can be asserted in the regression algorithm and accessible for the user
Should adjust the algo to _also_ use SetHoldings and will make the bug of the multiple accounting more clearer,

__
btw `CalculateOptionFee` already returns `quantity * orderPrice`, what about the contract multiplier?
__
the adjustments being made here for equity and options fees, should they also be for the main IB fee model?